crates/adapters/src/transport/pubsub/test.rs

```diff
@@ -0,0 +1,518 @@
+            topic.create(None, None).await.unwrap();
+
+            Self {
+                client,
+                publisher: topic.new_publisher(None),
+                topic,
+            }
+        })
+    }
+
+    fn with_auth(topic_id: &str) -> Self {
+        TOKIO.block_on(async {
+            info!("Creating test publisher using Application Default Credentials");
+
+            let config = ClientConfig::default().with_auth().await.unwrap();
+            //config.project_id = Some(PROJECT_ID.to_string());
+            //config.environment = Environment::Emulator(emulator.to_string());
+            let client = Client::new(config).await.unwrap();
+
+            // Create topic.
+            let topic = client.topic(topic_id);
+            if topic.exists(None).await.unwrap() {
+                info!("Deleting existing topic {topic_id}");
+
+                topic.delete(None).await.unwrap();
+            }
+
+            info!("Creating topic {topic_id}");
+            topic.create(None, None).await.unwrap();
+
+            let publisher = topic.new_publisher(None);
+
+            info!("Test publisher created");
+
+            Self {
+                client,
+                publisher,
+                topic,
+            }
+        })
+    }
+
+    fn create_subscription(&self, topic: &str, filter: &str) -> String {
+        let subscription_id = format!("sub-{}", &uuid::Uuid::new_v4());
+        TOKIO.block_on(async {
+            self.client
+                .create_subscription(
+                    &subscription_id,
+                    topic,
+                    SubscriptionConfig {
+                        enable_message_ordering: true,
+                        filter: filter.to_string(),
+                        retain_acked_messages: true,
+                        ..Default::default()
+                    },
+                    None,
+                )
+                .await
+                .unwrap();
+        });
+
+        subscription_id
+    }
+
+    fn send<T, LF>(&self, data: &[Vec<T>], ordered: bool, label_func: LF)
+    where
+        T: Clone + Serialize + Debug,
+        LF: Fn(&T) -> HashMap<String, String>,
+    {
+        TOKIO.block_on(async {
+            info!("sending");
+
+            for batch in data {
+                // Pub/Sub doesn't allow empty messages.
+                if batch.is_empty() {
+                    continue;
+                }
+                let mut writer = WriterBuilder::new()
+                    .has_headers(false)
+                    .from_writer(Vec::with_capacity(batch.len() * 32));
+
+                for val in batch.iter().cloned() {
+                    writer.serialize(val).unwrap();
+                }
+                writer.flush().unwrap();
+                let bytes = writer.into_inner().unwrap();
+
+                let message = PubsubMessage {
+                    data: bytes,
+                    ordering_key: if ordered {
+                        "1".to_string()
```

Comment:
what is 1? a field index?

Reply:
just an arbitrary string. Can be anything.